### PR TITLE
Add support for the UC_ATTR attribute

### PR DESCRIPTION
**Pull request by olejandro**

### CODE REVIEW

**Comment by olejandro** [comment]
@Antti-L  any chance I could use your expertise on the following issue?

Our TIMES-Ireland Model benchmark has the following entry:
```
SET UC_ATTR
/
'IE'.'UC_Max_SNSP'.LHS.UCN.DAYNITE
/;
```
As far as I can see in the docs, `DAYNITE` is not part of `UC_NAME` set, but since GAMS doesn't throw domain violation error while processing the corresponding input file, I guess, it must be. What am I missing?

**Comment by Antti-L** [comment]
Ah...  it is just the workaround for `UC_TSL` under VEDA.   Years ago, that seemed to provide an easier way to have VEDA support available for `UC_TSL`, as compared to requesting implementation of that new SET attribute in VEDA.  Therefore it was decided to implement such a workaround:  One can use `tslvl` as a UC_NAME in `UC_ATTR`, to define a constraint ts-dynamic (or "pseudo-dynamic") on the desired level.

**Comment by olejandro** [comment]
Ok, thanks! I've now added that possibility as well.

How about `N`, `CUM+`, `YES`, `ANNUL`, `FLO_COST`, `FLO_DELIV`, `FLO_SUB`, `FLO_TAX`, `NCAP_COST`, `NCAP_ITAX`, and `NCAP_ISUB`? Should all of them be also allowed?

**Comment by Antti-L** [comment]
> How about N, CUM+, YES, ANNUL, FLO_COST, FLO_DELIV, FLO_SUB, FLO_TAX, NCAP_COST, NCAP_ITAX, and NCAP_ISUB? Should all of them be also allowed?

Of these, YES is a documented modifier for users to use, and so must be allowed.  N, CUM+ and ANNUL are internal, and should best not be allowed. The rest are old-style user cost modifiers (not documented, but they do still work), and someone might use them.

**Comment by Antti-L** [comment]
I see the following column list:
`columns = [ "uc_n",  "description", "region", "reg_action", "ts_action", "uc_attr", "group_type",  "side",    ]`

I can see **reg_action** is apparently related to` R_EACH / R_SUM`, but I am curious about this "ts_action". I would have expected it to be related to `TS_EACH / TS_SUM`, but it doesn't seem to be, because in the code I see startswith("T_").  So, does this mean there is no timeslice action, and the `ts_action` is about periods, i.e. `T_EACH / T_SUM` ?

**Comment by olejandro** [comment]
Thanks @Antti-L. I've opened an issue to handle timeslice action. Will also rename the "actions" to avoid confusion between periods and timeslices. Unfortunately neither `TS_EACH` nor  `TS_SUM` appear in our benchmarks, so we don't have a test for it. If you happen to have a simple model that covers the case (and possibly more) and you are willing to share it, I'd be happy to include it in the benchmarks.

File: xl2times/transforms.py [hunk 1]
```diff
@@ -748,38 +724,112 @@ def generate_uc_properties(
...
+                    v.strip().upper()
+                    for v in row["uc_attr"].split(",")
+                    if v.strip().upper() in extended_uc_name
+                ],
+                axis=1,
+            )
+            user_constraints.loc[index, "uc_attr"] = user_constraints[index].apply(
+                lambda row: row["uc_attr"][-1] if row["uc_attr"] else None,
+                axis=1,
+            )
```
Comment (siddharth-krishna) [inline xl2times/transforms.py:832]:
```suggestion
            # Extend UC_NAME set with timeslice levels
            extended_uc_names = set(
                config.times_sets["UC_NAME"] + config.times_sets["TSLVL"]
            )
            uc_group_types = set(config.times_sets["UC_GRPTYPE"])

            def process_uc_attr(pair_str):
                items = [s.strip().upper() for s in pair_str.split(",")]
                group_types = [s for s in items if s in uc_group_types]
                uc_names = [s for s in items if s in extended_uc_names]
                if (
                    len(group_types) > 1
                    or len(uc_names) > 1
                    or len(group_types) + len(uc_names) != len(items)
                ):
                    raise ValueError(
                        f"uc_attr column value expected to be a pair (UC_GRPTYPE, UC_NAME/TSLVL) but got {pair_str}"
                    )
                return next(iter(group_types), None), next(iter(uc_names), None)

            group_types, uc_attrs = zip(
                *map(process_uc_attr, user_constraints.loc[index, "uc_attr"])
            )
            user_constraints.loc[index, "group_type"] = group_types
            user_constraints.loc[index, "uc_attr"] = uc_attrs
```

I tried to resolve the TODOs. :)

Comment (olejandro) [reply xl2times/transforms.py:832]: Great, thanks a lot!

**Pull request merged**